parasolr/solr/client.py

```diff
@@ -86,6 +96,58 @@ def _process_facet_counts(self, facet_counts: AttrDict) -> OrderedDict:
         return facet_counts
 
 
+class QueryResponse(BaseResponse):
+    """Thin wrapper to give access to Solr select responses.
+
+    Args:
+        response: A Solr query response
+    """
+
+    def __init__(self, response: Dict) -> None:
+        super().__init__(response)
+        # document list is contained with the "response" element
+        # in the json returned by solr
+        self.document_list = self.response.response
+
+    @property
+    def numFound(self) -> int:
+        return self.document_list.numFound
+
+    @property
+    def start(self) -> int:
+        return self.document_list.start
+
+    @property
+    def docs(self) -> List:
+        return self.document_list.docs
+
+
+class GroupedResponse(BaseResponse):
+    """Query response variant for grouped results.
+
+    Args:
+        response: A Solr query response
+    """
+
+    def __init__(self, response: Dict) -> None:
+        super().__init__(response)
+        # grouped response structure is structured as a dict
+        # first keyed on fieldname with number of matches, then a dict
+        # of group values and corresponding document list
+        self.grouped = self.response.grouped
+
+        # access grouped results at:
+        # self.grouped.fieldname.groups
+        # groups has
+        # will be a dict of field value, doclist
+
+    @property
+    def numFound(self) -> int:
+        # each field used for grouping has a total
+        # for the number of matches in that grouping
+        return sum(group["matches"] for group in self.grouped.values())
+
```

Comment:
There's no top level match count? I think you're right just wondering

Reply:
Maybe this could be a cached_property?

Reply:
Not in the examples I was looking at. You can get a count of groups, I think? But that's different from the number of records.  It's a little hard for me to imagine a use case for multiple groupings, this might double count things...  (I wondered about returning just the first group total, but this actually was easier to implement.)

Reply:
Ah, good thought on the cached property. I considered using that somewhere else (facet counts, I think?), but I don't think it's available for all of the versions of python we're currently supporting. Maybe when we do some app upgrades and drop the older versions of python we could sprinkle some of those in where it would help